compiler/rustc_interface/src/passes.rs

```diff
@@ -373,19 +373,23 @@ fn generated_output_paths(
 ) -> Vec<PathBuf> {
     let mut out_filenames = Vec::new();
     for output_type in sess.opts.output_types.keys() {
-        let file = outputs.path(*output_type);
+        let output = outputs.path(*output_type);
+        let file = output.as_path().to_path_buf();
         match *output_type {
             // If the filename has been overridden using `-o`, it will not be modified
             // by appending `.rlib`, `.exe`, etc., so we can skip this transformation.
             OutputType::Exe if !exact_name => {
                 for crate_type in sess.crate_types().iter() {
                     let p = filename_for_input(sess, *crate_type, crate_name, outputs);
-                    out_filenames.push(p);
+                    out_filenames.push(p.as_path().to_path_buf());
                 }
             }
             OutputType::DepInfo if sess.opts.unstable_opts.dep_info_omit_d_target => {
                 // Don't add the dep-info output when omitting it from dep-info targets
             }
+            OutputType::DepInfo if output == OutFileName::Stdout => {
+                // Don't add the dep-info output when it goes to stdout
```

Comment:
It's not clear to me why we wouldn't want to add this, can you elaborate?

Reply:
I added this check at some point because I found that if we added the dep-info output when it would go to stdout, then we wouldn't have a path for it. So the dep-info might look like
```
stdout: file1 file2 file3
```
which seems a bit odd to me. I can change it back.

Reply:
I'm not sure what the correct thing to do here is. You have given this more thought than me, so let's keep this if you prefer it.